packages/react-server-dom-webpack/src/ReactFlightDOMServerNode.js

```diff
@@ -99,12 +108,21 @@ function decodeReplyFromBusboy<T>(
           'the wrong assumption, we can easily fix it.',
       );
     }
+    pendingFiles++;
     const file = resolveFileInfo(response, name, filename, mimeType);
     value.on('data', chunk => {
       resolveFileChunk(response, file, chunk);
     });
     value.on('end', () => {
       resolveFileComplete(response, name, file);
+      pendingFiles--;
+      if (pendingFiles === 0) {
+        // Release any queued fields
+        for (let i = 0; i < queuedFields.length; i += 2) {
+          resolveField(response, queuedFields[i], queuedFields[i + 1]);
+        }
+        queuedFields.length = 0;
+      }
     });
   });
   busboyStream.on('finish', () => {
```

Comment:
wait do we need to delay this `close` on pendingFiles too?

Reply:
No because busboy has a similar built in queuing strategy to delay this call but not for fields.